pkg/ddc/jindo/dataset.go

```diff
@@ -29,6 +29,14 @@ import (
 	"k8s.io/client-go/util/retry"
 )
 
+// UpdateDatasetStatus updates the status of a dataset in the JindoEngine.
+// This function is primarily responsible for updating the phase and conditions of the dataset based on the given phase,
+// and updating the cache state based on the runtime status.
+// Parameters:
+//   - phase (datav1alpha1.DatasetPhase): The new phase to update the dataset to.
+//
+// Returns:
+//   - err (error): Returns an error if the update process fails, otherwise returns nil.
```

Comment:
I suggest using the following format to write multi-line comments in Go:

```golang
/*
UpdateDatasetStatus updates the phase and conditions of the dataset based on the given phase,
and updates the cache state based on the runtime status.

Parameters:
  - phase (datav1alpha1.DatasetPhase): The new phase to update the dataset to.

Returns:
  - err (error): Returns an error if the update process fails, otherwise returns nil.
*/

```